rcl/src/rcl/wait.c

```diff
@@ -64,6 +64,7 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   rcl_ret_t ret;
+  (void)ret;  // NO LINT
```

Comment:
Well, the variables in the different scopes only have the same name but are never used to pass information. They could easily have unique names. Therefore I think the smallest scope rule applies.